source/tutorial/gsfa/GSFA.rst

```diff
@@ -0,0 +1,427 @@
+features that preserve label similarities. More precisely, the algorithm
+utilize training graphs, in which the verticies are the samples and the
+edges represent similarities of the corresponding labels. Later, we use
+aquired low-dimensional representaion of the original data to train
+typical supervised learning algortihms.
+
+In this tutorial we will briefly exlpain the idea behind GSFA and
+specify the optimization task it solves. Moreover, we will show the
+effeciency of the GSFA compared to SVM on a toy example and introduce an
+approach that makes classsification task more interpretable.
+
+Table of Contents
+-----------------
+
+
+1. `Classification using GSFA`_
+
+2. `Idea behind GSFA`_
+
+3. `How the graphs are used for the training`_
+
+4. `GSFA Optimization Problem`_
+
+5. `Linear GSFA Algorithm`_
+
+
+--------------
+
+.. _Classification using GSFA:
+
+1. Classification using GSFA 
+----------------------------
+
+To show the benefits of the model and its efficiency we will solve the
+classification task that is based on
+`‘breast_cancer’ <https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_breast_cancer.html>`__
+dataset of sklearn library. More precisely, we will use SVM Classifier
+as a baseline method and later see the effect of preprocessing data via
+GSFA.
+
+The breast cancer dataset is a classic and very easy binary
+classification dataset. Features are computed from a digitized image of
+a fine needle aspirate (FNA) of a breast mass. They describe
+characteristics of the cell nuclei present in the image. It can be found
+on `UCI Machine Learning
+Repository <https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+%28Diagnostic%29>`__
+
+**Number of instances:** 569
+
+**Number of attributes:** 30 numeric, predictive attributes and the
+class
+
+**Attribute information:**
+
+-  radius (mean of distances from center to points on the perimeter)
+-  texture (standard deviation of gray-scale values)
+-  perimeter
+-  area
+-  smoothness (local variation in radius lengths)
+-  compactness (perimeter^2 / area - 1.0)
+-  concavity (severity of concave portions of the contour)
+-  concave points (number of concave portions of the contour)
+-  symmetry
+-  fractal dimension (“coastline approximation” - 1)
+
+*The mean, standard error, and “worst” or largest (mean of the three
+worst/largest values) of these features were computed for each image,
+resulting in 30 features. For instance, field 0 is Mean Radius, field 10
+is Radius SE, field 20 is Worst Radius.*
+
+**Class:**
+
+- WDBC-Malignant
+- WDBC-Benign
+
+
+First let’s import the usual data science modules and library
+`mdp <https://mdp-toolkit.github.io>`__. We use it for GSFA
+implementation and other data processing tools.
+
+
+	>>> %matplotlib inline
+	>>> import matplotlib.pyplot as plt
+	>>>
+	>>> from sklearn import (datasets, model_selection, metrics)
+	>>> import numpy as np
+	>>>
+	>>> import mdp 
+	>>> from sklearn.svm import SVC
+
```

Comment:
You are using plenty of blank lines, maybe too many? Please look into other rst files in this repository, e.g. https://raw.githubusercontent.com/mdp-toolkit/mdp-docs/master/source/tutorial/nodes.rst and try to adopt the style. Two consecutive blank lines are rare there, only at the end of high-level sections. I think three consecutive blank lines never occur, but I saw this in your rst (somewhere below).